packages/teleport/src/Player/DesktopPlayer.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import ProgressBar from './ProgressBar';
+import { PlayerClient, PlayerClientEvent } from 'teleport/lib/tdp';
+import { PngFrame, ClientScreenSpec } from 'teleport/lib/tdp/codec';
+import cfg from 'teleport/config';
+import { getAccessToken, getHostName } from 'teleport/services/api';
+import TdpClientCanvas from 'teleport/components/TdpClientCanvas';
+
+export const DesktopPlayer = ({
+  sid,
+  clusterId,
+}: {
+  sid: string;
+  clusterId: string;
+}) => {
+  const { playerClient, tdpCliOnPngFrame, tdpCliOnClientScreenSpec } =
+    useDesktopPlayer({
+      sid,
+      clusterId,
+    });
+
+  return (
+    <StyledPlayer>
+      <TdpClientCanvas
+        tdpCli={playerClient}
+        tdpCliOnPngFrame={tdpCliOnPngFrame}
+        tdpCliOnClientScreenSpec={tdpCliOnClientScreenSpec}
+        onContextMenu={() => true}
+        // overflow: 'hidden' is needed to prevent the canvas from outgrowing the container due to some weird css flex idiosyncracy.
+        // See https://gaurav5430.medium.com/css-flex-positioning-gotchas-child-expands-to-more-than-the-width-allowed-by-the-parent-799c37428dd6.
+        style={{ display: 'flex', flexGrow: 1, overflow: 'hidden' }}
+      />
+      <ProgressBarDesktop playerClient={playerClient} />
+    </StyledPlayer>
+  );
+};
+
+export const ProgressBarDesktop = (props: { playerClient: PlayerClient }) => {
+  const { playerClient } = props;
+
+  const [state, setState] = useState({
+    max: 500, // TODO: total time of the recording in ms
```

Comment:
Those need to be set to the actual parameters of the recording, and updated periodically so that the progress bar updates as the recording progresses.

Implemented in the follow-on PR: https://github.com/gravitational/webapps/pull/577